src/routes/auth.js

```diff
@@ -388,3 +390,50 @@ router.post(`/sudoTestCommand/:secret/makeadminforclient`, async (req, res) => {
         });
     }
 });
+
+router.post('/requestToken', async (req, res) => {
+    try {
+        const { jwt } = req.body;
+        const { clientId } = decode(jwt);
+        const client = await Client.findById(clientId);
+
+        if (!client) {
+            return res.status(400).json({
+                err: true,
+                msg: 'No client found',
+            });
+        }
+
+        verify(jwt, client.access_token, {
+            algorithms: ['HS256'],
+        });
+        let requestToken = makeid(64, true);
+        let alreadyExists = false;
+        do {
+            let tokenAlreadyExists = false;
+            rtoken.exists(requestToken.toString(), (_err, exists) => {
```

Comment:
I don't think it should work the way you expect it to. The callback function will be called asynchronously and `tokenAlreadyExists` might or might not be changed to reflect the actual value. I would suggest using the `await` construct so that the variable is set by the `rtoken.exists` call before checking its value in the following statement. Also please don't use so many variables for a simple task like checking duplicate entries. You can modify the code to look like something below

```js
var token, exists = false;
while(!exists){
    token = makeid(..);
    exists = await rtoken.exists(token);
}
```